version: '3'

services:
  db:
    image: postgres:13.3
    environment:
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
    networks:
      - symfony_network
    ports:
      - "5432:5432"
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 10s
      retries: 5

  php:
    build: ./
    networks:
      - symfony_network
    tty: true
    depends_on:
      db:
        condition: service_healthy
    environment:
      PHP_IDE_CONFIG: "serverName=localhost"
    volumes:
      - ./:/app
      - ./xdebug.ini:/usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini

  nginx:
    image: nginx:latest
    ports:
      - "80:80"
    networks:
      - symfony_network
    depends_on:
      - php
    volumes:
      - ./nginx.conf:/etc/nginx/conf.d/default.conf
      - ./:/app

networks:
  symfony_network:
    driver: bridge
